modules/tinymce/src/core/main/ts/Rtc.ts

```diff
@@ -257,23 +256,24 @@ const getRtcSetup = (editor: Editor): Optional<() => Promise<RtcRuntimeApi>> =>
     Optional.from(rtcPlugin.setup)
   );
 
-export const setup = (editor: Editor): Optional<Promise<boolean>> => {
+export const setup = (editor: Editor): Optional<() => Promise<boolean>> => {
   const editorCast = editor as RtcEditor;
   return getRtcSetup(editor).fold(
     () => {
       editorCast.rtcInstance = makePlainAdaptor(editor);
       return Optional.none();
     },
-    (setup) => Optional.some(
-      setup().then((rtcEditor) => {
-        editorCast.rtcInstance = makeRtcAdaptor(rtcEditor);
-        return rtcEditor.rtc.isRemote;
-      }, (err) => {
-        // We need to provide a noop adaptor on init failure since otherwise calls to hasUndo etc will continue to throw errors
-        editorCast.rtcInstance = makeNoopAdaptor();
-        return Promise.reject<boolean>(err);
-      })
-    )
+    (setup) => {
+      // We need to provide a noop adaptor while initializing since any call by the theme or plugins to say undoManager.hasUndo would throw errors
+      editorCast.rtcInstance = makeNoopAdaptor();
+
+      return Optional.some(
+        () => setup().then((rtcEditor) => {
```

Comment:
Just checking: what happened to the `err` here? I assume we no longer need it for some reason, but I couldn't figure out the logic.

Reply:
The errors gets handled later on. This specific error branch was added there to provide a noop rtc instance if an error was produced that is now replaced with the noop rtc instance that is always there while initializing but replaced with a real rtc instance once the editor has started.